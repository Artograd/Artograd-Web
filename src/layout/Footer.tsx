import React from 'react';
import { MainMenu, MainMenuButton, FlexSpacer, FlexCell } from '@epam/uui';
import { AdaptiveItemProps, MainMenuLogo } from '@epam/uui-components';
import { useTranslation } from 'react-i18next';

export const Footer = () => {
  const { t } = useTranslation();

  const getMenuItems = (): AdaptiveItemProps[] => {
    return [
      {
        id: 'logo',
        priority: 99,
        render: (p) => (
          <MainMenuLogo
            key={p.id}
            href={window.location.origin}
            logoUrl="/artograd.logo.svg"
          />
        ),
      },
      {
        id: 'flexSpacer1',
        priority: 100,
        render: (p) => <FlexSpacer key={p.id} />,
      },
      {
        id: 'copyrights',
        priority: 3,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            caption={t('global.layout.footer.copyright')}
          />
        ),
      },
      {
        id: 'flexSpacer2',
        priority: 100,
        render: (p) => <FlexSpacer key={p.id} />,
      },
      {
        id: 'privacyPolicy',
        priority: 3,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            caption={t('global.layout.footer.privatePolicyCta')}
          />
        ),
      },
      {
        id: 'cookiesPolicy',
        priority: 3,
        render: (p) => (
          <MainMenuButton
            key={p.id}
            caption={t('global.layout.footer.cookiePolicyCta')}
          />
        ),
      },
    ];
  };

  return (
    <FlexCell>
      <MainMenu items={getMenuItems()} />
    </FlexCell>
  );
};
