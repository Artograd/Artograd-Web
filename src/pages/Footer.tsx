import React from 'react';
import { BurgerButton, GlobalMenu, MainMenu, MainMenuAvatar, MainMenuButton, MainMenuIcon, FlexSpacer, FlexCell, DropdownMenuButton, DropdownMenuSplitter, DropdownMenuBody, Burger } from '@epam/uui';
import { Dropdown, AdaptiveItemProps, MainMenuLogo } from '@epam/uui-components';
import { ReactComponent as HelpIcon } from '@epam/assets/icons/common/notification-help-outline-24.svg';

export default function MainMenuBasicExample() {
    const renderBurger = (props: { onClose: () => void }) => (
        <>
            <BurgerButton
                href="/"
                caption="Training Catalog"
                onClick={ () => {
                    props.onClose && props.onClose();
                } }
            />
            <BurgerButton
                href="/"
                caption="Requests"
                onClick={ () => {
                    props.onClose && props.onClose();
                } }
            />
            <BurgerButton
                href="/"
                caption="Help"
                onClick={ () => {
                    props.onClose && props.onClose();
                } }
            />
            <BurgerButton
                href="/"
                caption="Settings"
                onClick={ () => {
                    props.onClose && props.onClose();
                } }
            />
            <BurgerButton
                href="/"
                caption="Log out"
                onClick={ () => {
                    props.onClose && props.onClose();
                } }
            />
        </>
    );

    const getMenuItems = (): AdaptiveItemProps[] => {
        return [
            {
                id: 'burger', priority: 100, collapsedContainer: true, render: (p) => <Burger key={ p.id } width={ 300 } renderBurgerContent={ renderBurger } />,
            },
            {
                id: 'logo',
                priority: 99,
                render: (p) => <MainMenuLogo key={ p.id } href={window.location.origin} logoUrl="/artograd.logo.svg" />,
            },
            { id: 'flexSpacer1', priority: 100, render: (p) => <FlexSpacer key={ p.id } /> },
            { id: 'copyrights', priority: 3, render: (p) => <MainMenuButton key={ p.id } caption="Â© 1993â€”2024 EPAM Systems. All Rights Reserved." /> },
            { id: 'flexSpacer2', priority: 100, render: (p) => <FlexSpacer key={ p.id } /> },
            { id: 'privacyPolicy', priority: 3, render: (p) => <MainMenuButton key={ p.id } href="/privace_policy" caption="Privacy policy" /> },
            { id: 'coockiesPolicy', priority: 3, render: (p) => <MainMenuButton key={ p.id } href="/cookies_policy" caption="Cookies policy" /> },
        ];
    };

    return (
        <FlexCell>
            <MainMenu items={ getMenuItems() } />
        </FlexCell>
    );
}