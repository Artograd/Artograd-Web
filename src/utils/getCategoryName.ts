import { CategoryItemType } from '../types';

export const categoryList: CategoryItemType[] = [
  { id: 0, name: 'tendersPage.newTender.categories.sculptures' },
  { id: 1, name: 'tendersPage.newTender.categories.mosaics' },
  { id: 2, name: 'tendersPage.newTender.categories.murals' },
  { id: 3, name: 'tendersPage.newTender.categories.graffiti' },
  { id: 4, name: 'tendersPage.newTender.categories.functionalArt' },
  {
    id: 5,
    name: 'tendersPage.newTender.categories.interactiveInstallations',
  },
  { id: 6, name: 'tendersPage.newTender.categories.botanicalArt' },
  { id: 7, name: 'tendersPage.newTender.categories.waterFeatures' },
  { id: 8, name: 'tendersPage.newTender.categories.themedGardens' },
  { id: 9, name: 'tendersPage.newTender.categories.recycled' },
];

export const getCategoryName = (id: string) => {
  return categoryList.find((category) => category.id === Number(id));
};
