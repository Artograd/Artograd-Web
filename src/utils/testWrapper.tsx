import { renderWithContextAsync } from '@epam/uui-test-utils';
import { MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { store } from '../store/store';

export const testWrapper = async ({
  component,
  history,
}: {
  component: React.JSX.Element;
  history: MemoryHistory<unknown>;
  
}) => {
  return renderWithContextAsync(<Provider store={store}><Router history={history}>{component}</Router></Provider>);
};
