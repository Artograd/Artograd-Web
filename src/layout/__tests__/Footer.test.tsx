import { Footer } from '../Footer';
import { screen, fireEvent } from '@epam/uui-test-utils';
import { createMemoryHistory } from 'history';
import { testWrapper } from '../../utils/testWrapper';

const menuLinks = [
  {
    linkName: 'Privacy policy',
    url: '/policy',
  },
  {
    linkName: 'Cookies policy',
    url: '/cookie',
  },
];

describe('Layout footer', () => {
  const history = createMemoryHistory();
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', async () => {
    const component = await testWrapper({ component: <Footer />, history });

    expect(component).toMatchSnapshot();
  });

  menuLinks.map((link) => {
    test('redirect on menu link click', async () => {
      await testWrapper({ component: <Footer />, history });
      fireEvent.click(screen.getByText(link.linkName));

      expect(history.location.pathname).toBe(link.url);
    });
  });
});
