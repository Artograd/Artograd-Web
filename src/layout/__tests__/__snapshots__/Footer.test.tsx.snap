// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout footer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="BULoA8 uui-main_menu uui-mainmenu-container aqkY9D"
        >
          <div
            class="bziidR HXwiKY"
          >
            <div
              class="uui-flex-row x8mfAd q3nKl2"
            >
              <div>
                <a
                  aria-disabled="false"
                  class="YHd3ss uui-anchor uui-enabled -clickable lhoOmi"
                  href="http://localhost"
                  role="link"
                  tabindex="0"
                >
                  <img
                    alt="Main Menu Logo"
                    class="qzo1nb"
                    src="artograd.logo.svg"
                  />
                </a>
              </div>
              <div
                class="bTGcnV"
              />
              <button
                class="penmb- uui-button-box uui-enabled RL38By _8kGjpz"
                tabindex="-1"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  © 1993—2024 EPAM Systems. All Rights Reserved.
                </div>
              </button>
              <div
                class="bTGcnV"
              />
              <button
                class="penmb- uui-button-box uui-enabled -clickable RL38By _8kGjpz footerLink"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Privacy policy
                </div>
              </button>
              <button
                class="penmb- uui-button-box uui-enabled -clickable RL38By _8kGjpz footerLink"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Cookies policy
                </div>
              </button>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
      style="min-width: 0; flex-basis: 0px;"
    >
      <nav
        class="BULoA8 uui-main_menu uui-mainmenu-container aqkY9D"
      >
        <div
          class="bziidR HXwiKY"
        >
          <div
            class="uui-flex-row x8mfAd q3nKl2"
          >
            <div>
              <a
                aria-disabled="false"
                class="YHd3ss uui-anchor uui-enabled -clickable lhoOmi"
                href="http://localhost"
                role="link"
                tabindex="0"
              >
                <img
                  alt="Main Menu Logo"
                  class="qzo1nb"
                  src="artograd.logo.svg"
                />
              </a>
            </div>
            <div
              class="bTGcnV"
            />
            <button
              class="penmb- uui-button-box uui-enabled RL38By _8kGjpz"
              tabindex="-1"
              type="button"
            >
              <div
                class="uui-caption"
              >
                © 1993—2024 EPAM Systems. All Rights Reserved.
              </div>
            </button>
            <div
              class="bTGcnV"
            />
            <button
              class="penmb- uui-button-box uui-enabled -clickable RL38By _8kGjpz footerLink"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Privacy policy
              </div>
            </button>
            <button
              class="penmb- uui-button-box uui-enabled -clickable RL38By _8kGjpz footerLink"
              tabindex="0"
              type="button"
            >
              <div
                class="uui-caption"
              >
                Cookies policy
              </div>
            </button>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
